cl/corpus_importer/tasks.py

```diff
@@ -1215,7 +1225,167 @@ def make_docket_by_iquery(
         d,
         tag_names,
         add_to_solr=True,
+        from_iquery_scrape=from_iquery_scrape,
+    )
+
+
+@retry((requests.Timeout, PacerLoginException), tries=3, delay=0.25, backoff=1)
+def query_iquery_page(
+    court_id: str, pacer_case_id: int
+) -> bool | dict[str, Any]:
+    """Query the iquery page for a given PACER case ID.
+
+    :param court_id: A CL court ID where we'll look things up.
+    :param pacer_case_id: The Pacer Case ID to lookup.
+    :return: False if not valid report, otherwise the report.data.
+    """
+
+    cookies = get_or_cache_pacer_cookies(
+        "pacer_scraper",
+        settings.PACER_USERNAME,
+        password=settings.PACER_PASSWORD,
+    )
+    s = ProxyPacerSession(
+        cookies=cookies,
+        username=settings.PACER_USERNAME,
+        password=settings.PACER_PASSWORD,
     )
+    report = CaseQuery(map_cl_to_pacer_id(court_id), s)
+    report.query(pacer_case_id)
+    if not report.data:
+        logger.info(
+            "No valid data found in iquery page for %s.%s",
+            court_id,
+            pacer_case_id,
+        )
+        return False
+    return report.data
+
+
+@app.task(
+    bind=True,
+    autoretry_for=(RedisConnectionError,),
+    max_retries=3,
+    interval_start=1 * 60,
+    interval_step=5 * 60,
+    ignore_result=True,
+)
+def iquery_pages_probing(
+    self: Task,
+    court_id: str,
+    testing: bool = False,
+) -> None:
+    """
+    Using the iquery endpoint, to perform forward probing and retrieve the
+    highest watermark we can scrape.
+
+    :param self: The celery task
+    :param court_id: A CL court ID where we'll look things up.
+    :param testing: A boolean indicating whether this was called from tests.
+    :return: None
+    """
+
+    r = get_redis_interface("CACHE")
+    probe_iteration = 1
+    latest_match = 0
+    time_out_counter = 0
+    consecutive_empty_probes = 0
+    iquery_pacer_case_id_final = int(
+        r.hget("iquery_pacer_case_id_final", court_id) or 0
+    )
+
+    # Probe limit e.g: 9 probe iterations -> 256
+    probe_limit = 2 ** (settings.IQUERY_PROBE_ITERATIONS - 1)
+    court_probe_cycle_no_hits = r.hincrby(
+        "court_probe_cycle_no_hits", court_id, 1
+    )
+    reports_data = []
+    while probe_iteration <= settings.IQUERY_PROBE_ITERATIONS:
+        if time_out_counter >= 3:
+            logger.warning(
+                "Aborting probing task for %s after 3 consecutive iterations "
+                "that timed out.",
+                court_id,
+            )
+            r.setex(
+                f"court_wait:{court_id}",
+                settings.IQUERY_COURT_BLOCKED_WAIT,
+                2,
+            )
+            break
+
+        probe_iteration, pacer_case_id_to_lookup = compute_next_binary_probe(
```

Comment:
It'd be simpler not to mess around with probe_iteration in here, no? Why not just increment that outside the function?

Reply:
Yes, this is better. Now `compute_next_binary_probe`  only returns `pacer_case_id_to_lookup`